services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -433,24 +436,77 @@ export class WorkflowService {
       projectIds,
     );
 
-    const totalWorkflowsCount = await this.workflowRuntimeDataRepository.count(
-      {
-        where: query.where,
-      },
-      projectIds,
+    const { workflowDefinitionIds, statuses, assigneeIds, includeUnassigned } = {
+      workflowDefinitionIds:
+        query.where.workflowDefinitionId &&
+        typeof query.where.workflowDefinitionId === 'object' &&
+        'in' in query.where.workflowDefinitionId &&
+        Array.isArray(query.where.workflowDefinitionId.in)
```

Comment:
Pretty sure this code can be cut down to `Array.isArray(query.where.workflowDefinitionId.in)` as it also checks the previous conditions.